src/modules/flight_mode_manager/tasks/AutoFollowTarget/FlightTaskAutoFollowTarget.hpp

```diff
@@ -105,111 +135,214 @@ class FlightTaskAutoFollowTarget : public FlightTask
 
-	// Lowpass filters for smoothingtarget position because it's used for setpoint generation
-	AlphaFilter<matrix::Vector3f> _target_position_filter;
+	// Last target estimator timestamp to handle timeout filter reset
+	uint64_t _last_valid_target_estimator_timestamp{0};
+
+	// Second Order Filter to calculate kinematically feasible target position
+	SecondOrderReferenceModel<matrix::Vector3f> _target_position_velocity_filter;
 
-	// Lowpass filter for smoothing the offset vector and have more dynamic shots when target changes direction
-	AlphaFilter<matrix::Vector2f> _offset_vector_filter;
+	// Internally tracked Follow Target characteristics, to allow RC control input adjustments
+	float _follow_distance{8.0f}; // [m]
+	float _follow_height{10.0f}; // [m]
+	float _follow_angle_rad{0.0f}; // [rad]
 
-	// Lowpass filter assuming  values 0-1, for avoiding big steps in velocity feedforward
-	AlphaFilter<float> _velocity_ff_scale;
+	// Tracked estimate of target's course (where velocity vector is pointing). Initialized as North direction
+	float _target_course_rad{0.0f};
+	// Tracked orbit angle setpoint that gets calculated from Jerk-limited trajectory. Initialized in North direction.
+	float _orbit_angle_setpoint_rad{0.0f};
 
-	// NOTE: If more of these internal state variables come into existence, it
-	// would make sense to create an internal state machine with a single enum
-	bool _emergency_ascent = false;
+	// Angular Jerk limited orbit angle setpoint trajectory generator
+	VelocitySmoothing _orbit_angle_traj_generator;
 
-	DEFINE_PARAMETERS(
+	// Variable to remember the home position (take-off point) z coordinate, where the follow height will be measured from
+	float _home_position_z;
+
+	DEFINE_PARAMETERS_CUSTOM_PARENT(
+		FlightTask,
 		(ParamInt<px4::params::MAV_SYS_ID>) _param_mav_sys_id,
 		(ParamInt<px4::params::MAV_COMP_ID>) _param_mav_comp_id,
-		(ParamFloat<px4::params::NAV_MIN_FT_HT>) _param_nav_min_ft_ht,
-		(ParamFloat<px4::params::NAV_FT_DST>) _param_nav_ft_dst,
-		(ParamInt<px4::params::NAV_FT_FS>) _param_nav_ft_fs,
-		(ParamInt<px4::params::NAV_FT_ALT_M>) _param_nav_ft_alt_m,
-		(ParamInt<px4::params::NAV_FT_GMB_M>) _param_nav_ft_gmb_m,
-		(ParamInt<px4::params::NAV_FT_DELC>) _param_nav_ft_delc
+		(ParamFloat<px4::params::FLW_TGT_HT>) _param_flw_tgt_ht,
+		(ParamFloat<px4::params::FLW_TGT_DST>) _param_flw_tgt_dst,
+		(ParamInt<px4::params::FLW_TGT_FP>) _param_flw_tgt_fp,
+		(ParamInt<px4::params::FLW_TGT_ALT_M>) _param_flw_tgt_alt_m,
+		(ParamFloat<px4::params::FLW_TGT_MAX_VEL>) _param_flw_tgt_max_vel
 	)
 
+	// uORB topics to subscribe to
 	uORB::Subscription _follow_target_estimator_sub{ORB_ID(follow_target_estimator)};
 
-	uORB::PublicationMulti<follow_target_status_s> _follow_target_status_pub{ORB_ID(follow_target_status)};
-	uORB::PublicationMulti<gimbal_manager_set_attitude_s> _gimbal_manager_set_attitude_pub{ORB_ID(gimbal_manager_set_attitude)};
+	// uORB topics to publish
```

Comment:
`uORB::Publication`, I guess it's clear enough

Reply:
Yep